server/src/main/java/org/elasticsearch/discovery/ConfiguredHostsResolver.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.discovery;
+
+import org.elasticsearch.common.transport.TransportAddress;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+public interface ConfiguredHostsResolver {
```

Comment:
I'm not sure this should be a top-level class. It feels more like a helper to PeerFinder.

Reply:
Ok, see 8b605ada5cfcd6a76704fceb2fc38aceda86cd2a